### PR TITLE
v2.7 release note

### PR DESCRIPTION
**Pull request by J-shang**

### CODE REVIEW

File: docs/source/release.rst [hunk 1]
```diff
@@ -5,6 +5,54 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+* [Improvement] TPE and random tuners will not generate duplicate hyperparameters anymore.
+* [Improvement] Most Python APIs now have type annotations.
+
+Neural Architecture Search
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+* Jointly search for architecture and hyper-parameters: ValueChoice in evaluator (`doc <https://nni.readthedocs.io/en/v2.7/reference/nas/search_space.html#valuechoice>`__)
+* Support composition (transformation) of one or several value choices (`doc <https://nni.readthedocs.io/en/v2.7/reference/nas/search_space.html#valuechoice>`__)
+* Enhanced Cell API (``merge_op``, preprocessor, postprocessor) (`doc <https://nni.readthedocs.io/en/v2.7/reference/nas/search_space.html#cell>`__)
+* Repeat depth now allows a ValueChoice (`doc <https://nni.readthedocs.io/en/v2.7/reference/nas/search_space.html#repeat>`__)
```
Comment (QuanluZhang) [inline docs/source/release.rst:28]: The argument `depth` in the `Repeat` API allows ValueChoice

File: docs/source/release.rst [hunk 2]
```diff
@@ -5,6 +5,54 @@
...
+* Support loading ``state_dict`` between sub-net and super-net (`doc <https://nni.readthedocs.io/en/v2.7/reference/nas/others.html#nni.retiarii.utils.original_state_dict_hooks>`__, `example in spos <https://nni.readthedocs.io/en/v2.7/reference/nas/strategy.html#spos>`__)
+* Support BN fine-tuning and evaluation in SPOS example (`doc <https://nni.readthedocs.io/en/v2.7/reference/nas/strategy.html#spos>`__)
+* *Experimental* Model hyper-parameter choice (`doc <https://nni.readthedocs.io/en/v2.7/reference/nas/search_space.html#modelparameterchoice>`__)
+* *Preview* Lightning implementation for Retiarii including DARTS, ENAS, ProxylessNAS and RandomNAS (`example usage <https://github.com/microsoft/nni/blob/v2.7/test/ut/retiarii/test_oneshot.py>`__)
+* *Preview* A search space hub that contains 10 search spaces (`code <https://github.com/microsoft/nni/tree/v2.7/nni/retiarii/hub>`__)
+
+Model Compression
+^^^^^^^^^^^^^^^^^
+* Pruning V2 is promoted as default pruning framework, old pruning is legacy and keeps for few releases.(`doc <https://nni.readthedocs.io/en/v2.7/reference/compression/pruner.html>`__)
```
Comment (QuanluZhang) [inline docs/source/release.rst:37]: few -> a few

**Review by QuanluZhang — APPROVED** [review]

**Review by JiahangXu — APPROVED** [review]

File: docs/source/release.rst (hunk 1 above)
Comment (J-shang) [reply docs/source/release.rst]: fixed

**Comment by matluster** [comment]
Sorry I merged this PR by mistake -- maybe something else was in my mind, and the approval button and merge button just looked too similar (both green) on mobile app.

If someone has any other comments or suggestions, let's refine the release note in another pull request.

I'm glad that I didn't delete anything by mistake. 😞

**Pull request merged**